ddtrace/llmobs/decorators.py

```diff
@@ -48,17 +79,32 @@ async def wrapper(*args, **kwargs):
 
             else:
 
+                @wraps(func)
+                def generator_wrapper(*args, **kwargs):
+                    if not LLMObs.enabled:
+                        log.warning(SPAN_START_WHILE_DISABLED_WARNING)
+                        yield from func(*args, **kwargs)
```

Comment:
nice, so `yield from` already handles `throw`, `send`, `GeneratorExit`, `StopIteration`?

Reply:
yeah, it does (from some testing i've done) for sync generators (`yield from` can only be used in sync cases). the `async` versions are a bit different, will push some tests & fixes up today trying to handle `async` generators better for those cases.